src/Routing/RouteBuilder.php

```diff
@@ -239,21 +252,31 @@ public function resources($name, $options = [], $callback = null)
         $options += [
             'connectOptions' => [],
             'id' => static::ID . '|' . static::UUID,
-            'only' => ['index', 'update', 'create', 'view', 'delete'],
+            'only' => [],
             'actions' => [],
+            'map' => [],
         ];
-        $options['only'] = (array)$options['only'];
-        $connectOptions = $options['connectOptions'];
 
-        $urlName = Inflector::underscore($name);
+        foreach ($options['map'] as $k => $mapped) {
+            $options['map'][$k] += ['method' => 'GET', 'path' => $k, 'action' => ''];
+        }
 
         $ext = null;
         if (!empty($options['_ext'])) {
             $ext = $options['_ext'];
         }
 
-        foreach (static::$_resourceMap as $method => $params) {
-            if (!in_array($method, $options['only'], true)) {
+        $connectOptions = $options['connectOptions'];
+        $urlName = Inflector::underscore($name);
```

Comment:
Not too thrilled about this underscore inflection. I personally prefer using dash instead of underscore as word separator in url.  Can this be made configurable using `urlName` key in options?